src/api/types/TypeRegistry.js

```diff
@@ -52,10 +53,31 @@ define(['./Type'], function (Type) {
      * @memberof module:openmct.TypeRegistry#
      */
     TypeRegistry.prototype.addType = function (typeKey, typeDef) {
+        this.standardizeType(typeDef);
         this.types[typeKey] = new Type(typeDef);
     };
 
     /**
+     * Takes a typeDef, standardizes it, and logs warnings about unsupported
+     * usage.
+     * @private
+     */
+    TypeRegistry.prototype.standardizeType = function (typeDef) {
+        if (typeDef.label) {
```

Comment:
Suggest `typeDef.hasOwnProperty("label")` so `{label: ""}` can't sneak past 

(or  `Object.prototype.hasOwnProperty.call(typeDef, "label")` so `{ hasOwnProperty: "I'm a jerk." }` doesn't cause an exception, but if you're that much of a jerk maybe you deserve an exception.)